src/freenet/clients/http/N2NTMToadlet.java

```diff
@@ -285,6 +289,14 @@ public static void createN2NTMSendForm(HTMLNode pageNode, boolean advancedMode,
 		}
 		messageForm.addChild("textarea", new String[] { "id", "name", "rows",
 				"cols" }, new String[] { "n2ntmtext", "message", "8", "74" });
+		if (fProxyJavascriptEnabled) {
+			HTMLNode letterCountInfo = messageForm.addChild("div", new String[] {"class"}, new String[] {"n2ntmcountouter"});
+			letterCountInfo.addChild("#", NodeL10n.getBase().getString("N2NTMToadlet.remainingLetters"));
+			letterCountInfo.addChild("span", new String[] {"id"}, new String[] {"n2ntmcount"});
+			letterCountInfo.addChild("script", new String[] {"type"}, new String[] {"text/javascript"})
```

Comment:
Add classes to some of the elements and bind the listeners to all instances of elements with the right classes. In order to use it multiple times on a single page, add one “top-level” class to some container that contains both the input field and the counter display, then add classes to both the the input field and the counter display itself. Searching for the all elements with the top-level class and then searching for the elements with the classes below that gives you an easy way to restrict your listeners and actions so that you don’t have to mess around with single IDs.